packages/icons/webpack.maven.js

```diff
@@ -0,0 +1,125 @@
+const path = require('path');
+const fs = require('fs');
+const ZipPlugin = require('zip-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const request = require('request');
+const crypto = require('crypto');
+
+const release = !!process.env['TALEND_RELEASE'];
+const mvnConfig = {
+  groupId: 'org.talend.ui',
+  artifactId: 'talend-icon',
+  version: require(path.join(__dirname, 'package.json')).version + (release ? '' : '-SNAPSHOT'),
+  packaging: 'jar',
+  repository: process.env['NEXUS_BASE'] || 'https://artifacts-zl.talend.com/nexus/'
+};
+
+module.exports = {
+  output: {
+		filename: 'maven.js',
+		path: path.join(__dirname, 'dist/maven')
+	},
+	plugins: [
+    new CopyWebpackPlugin([{ from: 'src/svg' }]),
+    new ZipPlugin({
+      filename: `${mvnConfig.artifactId}-${mvnConfig.version}`,
+      extension: mvnConfig.packaging,
+      pathPrefix: 'icons',
+      include: [/\.svg$/],
+    }),
+    {
+      apply: function (compiler) {
```

Comment:
Could you separate that function into sub-function to make it easier to understand.